services/slam/slam_test.go

```diff
@@ -1,119 +0,0 @@
-// in order to be run. It utilizes the internal package located in slam_test_helper.go to access
-// certain exported functions which we do not want to make available to the user.
-package slam_test
-
-import (
-	"context"
-	"errors"
-	"testing"
-
-	"go.viam.com/test"
-
-	"go.viam.com/rdk/components/generic"
-	"go.viam.com/rdk/registry"
-	"go.viam.com/rdk/resource"
-	"go.viam.com/rdk/services/slam"
-	"go.viam.com/rdk/testutils/inject"
-	rdkutils "go.viam.com/rdk/utils"
-)
-
-var (
-	testSvcName1 = "svc1"
-	testSvcName2 = "svc2"
-)
-
-func TestRegisteredReconfigurable(t *testing.T) {
-	s := registry.ResourceSubtypeLookup(slam.Subtype)
-	test.That(t, s, test.ShouldNotBeNil)
-	r := s.Reconfigurable
-	test.That(t, r, test.ShouldNotBeNil)
-}
-
-func TestWrapWithReconfigurable(t *testing.T) {
-	svc := &mock{name: testSvcName1}
-	reconfSvc1, err := slam.WrapWithReconfigurable(svc, resource.Name{})
-	test.That(t, err, test.ShouldBeNil)
-
-	_, err = slam.WrapWithReconfigurable(nil, resource.Name{})
-	test.That(t, err, test.ShouldBeError, slam.NewUnimplementedInterfaceError(nil))
-
-	reconfSvc2, err := slam.WrapWithReconfigurable(reconfSvc1, resource.Name{})
-	test.That(t, err, test.ShouldBeNil)
-	test.That(t, reconfSvc2, test.ShouldEqual, reconfSvc1)
-}
-
-func TestReconfigurable(t *testing.T) {
-	actualSvc1 := &mock{name: testSvcName1}
-	reconfSvc1, err := slam.WrapWithReconfigurable(actualSvc1, resource.Name{})
-	test.That(t, err, test.ShouldBeNil)
-	test.That(t, reconfSvc1, test.ShouldNotBeNil)
-
-	actualArm2 := &mock{name: testSvcName2}
-	reconfSvc2, err := slam.WrapWithReconfigurable(actualArm2, resource.Name{})
-	test.That(t, err, test.ShouldBeNil)
-	test.That(t, reconfSvc2, test.ShouldNotBeNil)
-	test.That(t, actualSvc1.reconfCount, test.ShouldEqual, 0)
-
-	err = reconfSvc1.Reconfigure(context.Background(), reconfSvc2)
-	test.That(t, err, test.ShouldBeNil)
-	test.That(t, reconfSvc1, test.ShouldResemble, reconfSvc2)
-	test.That(t, actualSvc1.reconfCount, test.ShouldEqual, 1)
-
-	err = reconfSvc1.Reconfigure(context.Background(), nil)
-	test.That(t, err, test.ShouldNotBeNil)
-	test.That(t, err, test.ShouldBeError, rdkutils.NewUnexpectedTypeError(reconfSvc1, nil))
-}
-
-func TestFromRobot(t *testing.T) {
-	testSvcName := "test"
-	svc1 := &mock{name: testSvcName}
-	r := &inject.Robot{
-		ResourceByNameFunc: func(name resource.Name) (interface{}, error) {
-			if name.ShortName() == testSvcName {
-				return svc1, nil
-			}
-			return nil, errors.New("not found")
-		},
-	}
-
-	t.Run("successfully find service", func(t *testing.T) {
-		svc2, err := slam.FromRobot(r, testSvcName)
-		test.That(t, err, test.ShouldBeNil)
-		test.That(t, svc2, test.ShouldNotBeNil)
-	})
-
-	t.Run("fail to find service", func(t *testing.T) {
-		svc2, err := slam.FromRobot(r, "garbage")
-		test.That(t, err, test.ShouldNotBeNil)
-		test.That(t, svc2, test.ShouldBeNil)
-	})
-}
-
-func TestDoCommand(t *testing.T) {
```

Comment:
[q] Why is this being removed? Do we not need to ensure the slam service implements `DoCommand`?

Reply:
every service tested the same thing that ends up using common code that is tested elsewhere. now that things are being embedded, I removed these copy pastes. The test did not really test anything